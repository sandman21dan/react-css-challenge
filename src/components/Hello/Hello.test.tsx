import * as React from 'react';
import { Hello } from './index';
import { shallow } from 'enzyme';
import * as renderer from 'react-test-renderer';

const incrementFn = jest.fn();
const decrementFn = jest.fn();

const component = (<Hello
  name="Test"
  enthusiasmLevel={3}
  onIncrement={incrementFn}
  onDecrement={decrementFn}
/>);

afterEach(() => {
  jest.resetAllMocks();
});

it('renders correctly', () => {
  const tree = renderer.create(component);
  expect(tree).toMatchSnapshot();
});

it('fires the increment function', () => {
  const wrapper = shallow(component);
  wrapper.find('button').first().simulate('click');

  expect(incrementFn).toHaveBeenCalled();
  expect(decrementFn).not.toHaveBeenCalled();
});

it('fires the decrement function', () => {
  const wrapper = shallow(component);
  wrapper.find('button').last().simulate('click');

  expect(decrementFn).toHaveBeenCalled();
  expect(incrementFn).not.toHaveBeenCalled();
});
