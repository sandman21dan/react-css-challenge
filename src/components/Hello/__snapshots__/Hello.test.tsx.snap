// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="hello"
>
  <p>
    Hello 
    Test
     
    3
    !
  </p>
  <div>
    <button
      onClick={[MockFunction]}
    >
      +
    </button>
    <button
      onClick={[MockFunction]}
    >
      -
    </button>
  </div>
</div>
`;
